autotest/gcore/tiff_write.py

```diff
@@ -7331,6 +7331,36 @@ def test_tiff_write_jpeg_incompatible_of_paletted():
 
 
 ###############################################################################
+# Test blocksize overriding while creating (internal) overviews
+# on a newly created dataset
+
```

Comment:
@tbonfort @rouault remember that since the project is using pytest, you can easily turn the blockSizes into separate isolated test cases.

```
@pytest.mark.parameterize("blockSize", [64, 256])
def test_tiff_write_186_blocksize(blockSize):
    src_ds = gdal.Open('../gdrivers/data/utm.tif')
    ...
```

Reply:
Or

```
@pytest.mark.parameterize(opts", [dict(GDAL_TIFF_OVR_BLOCKSIZE="64"), dict(GDAL_TIFF_OVR_BLOCKSIZE="256", GDAL_NUM_THREADS="8")]
def test_tiff_write_186_blocksize(opts):
    ...
    with gdaltest.config_options(opts):
            ds.BuildOverviews('AVERAGE', overviewlist=[2])
    ...
```

Reply:
yes, I thought at that at one moment, but forget it. Just fixed that